src/test/util_tests.cpp

```diff
@@ -130,6 +130,14 @@ BOOST_AUTO_TEST_CASE(util_ParseHex)
     result = ParseHex(" 89 34 56 78");
     BOOST_CHECK(result.size() == 4 && result[0] == 0x89 && result[1] == 0x34 && result[2] == 0x56 && result[3] == 0x78);
 
+    // Embedded null is treated as end
```

Comment:
I hope this is temporary I think this is very bad behavior, in general. Embedded null should be regarded as invalid characters like any other.

Reply:
Yes, I was planning to fix this bug after #https://github.com/bitcoin/bitcoin/pull/23595, but for now I am not going to change behaviour in this refactoring pull.

Reply:
Agree. I don't mean that, it's good that attention is brought to this in this way.